synapse/storage/databases/main/room.py

```diff
@@ -1274,6 +1274,29 @@ async def is_partial_state_room(self, room_id: str) -> bool:
 
         return entry is not None
 
+    @cachedList(cached_method_name="is_partial_state_room", list_name="room_ids")
+    async def is_partial_state_room_batched(
+        self, room_ids: StrCollection
+    ) -> Mapping[str, bool]:
+        """Checks if the given rooms have partial state.
+
+        Returns true for "partial-state" rooms, which means that the state
+        at events in the room, and `current_state_events`, may not yet be
+        complete.
+        """
+
+        rows: List[Dict[str, str]] = await self.db_pool.simple_select_many_batch(
+            table="partial_state_rooms",
+            column="room_id",
+            iterable=room_ids,
+            retcols=("room_id",),
+            desc="is_partial_state_room_batched",
+        )
+        partial_state_rooms = {
+            room_id for row_dict in rows for room_id in row_dict.values()
+        }
```

Comment:
I think this is much simpler:

```suggestion
        partial_state_rooms = {
            row_dict["room_id"] for row_dict in rows
        }
```